app/controllers/posts_controller.rb

```diff
@@ -2,12 +2,9 @@ class PostsController < ApplicationController
   include PostsHelper
 
   def index
-    @user = User.find_by_id(params[:user_id])
-    if @user.nil?
-      redirect_to '/users/invalid_user'
-    else
-      @posts = @user.posts
-    end
+    @user = User.includes(posts: { comments: :author }).find_by(id: params[:user_id])
+    redirect_to '/users/invalid_user' if @user.nil?
+    @posts = @user.posts
   end
```

Comment:
- [OPTIONAL] A correct way to eliminate N+1 queries when getting all posts for a User would be to write your code this way:

```ruby
@posts = @user.posts.includes(:comments)
``` 
instead of :
```ruby
@posts = @user.posts
``` 

It would be nice to implement it this way. 👍